cmd/determine-rebottle-runners.rb

```diff
@@ -82,6 +82,9 @@ def determine_rebottle_runners
       end
     end.compact
 
-    puts "::set-output name=runners::#{runners.to_json}"
+    github_output = ENV.fetch("GITHUB_OUTPUT") { raise "GITHUB_OUTPUT is not defined" }
+    File.open(github_output, "a") do |f|
+      f.puts("runners=#{runners.to_json}")
+    end
```

Comment:
Might be nicer to have a `GitHub.` helper in Homebrew/brew for this given we'll need it in a few places.

In general I wonder whether this command/file would be better placed in Homebrew/bre, too.

Reply:
It only makes sense and works within the context of Homebrew/core CI and does not work for non-Homebrew taps. It is very closely coupled with the rebottle workflow in this repository.

Reply:
Moving to Homebrew/brew arguably makes it harder, not easier, to maintain.

Reply:
@Bo98 This seems trivially easy to accidentally break from Homebrew/brew changes. At a minimum it'd be good to move more of this logic into Homebrew/brew, add unit tests and have them called from here. Stuff like `tags = formula.bottle_specification.collector.tags` is very much an implementation detail that feels very weird to be relying on here.

Reply:
That's only a few line. I think it would be nice to have better API for that because the remaining 90% of this file is specific to the names we have in the workflow, the use of CI provider, how we name our runners, what versions of macOS we build for in this repo, what versions of macOS are ephemeral, the use of Linux self-hosted vs GitHub, and the timeout values we desire for each.

Reply:
@Bo98 if even a single file here can be moved to Homebrew/brew: I'm all for it 😁. You've convinced me that having this command in this repository does make more sense, though. Also, better in a command like this than in e.g. a workflow file

Reply:
To me there's no reason why there shouldn't be a public API for knowing what bottle tags were built for a formula. I think if we solve that, then the file should be pretty much all public API that would go through deprecation if changed.

Reply:
Well besides the CLI parser public API debate which has never been settled.

Reply:
> To me there's no reason why there shouldn't be a public API for knowing what bottle tags were built for a formula.

If it's marked as a public API and documented 👍🏻 from me for usage.

> Well besides the CLI parser public API debate which has never been settled.

IMO: this is also clearly a public API and should be documented accordingly.